astropy/nddata/mixins/ndslicing.py

```diff
@@ -105,8 +105,9 @@ def _slice_uncertainty(self, item):
             return None
         try:
             return self.uncertainty[item]
-        except TypeError:
+        except (TypeError, KeyError):
```

Comment:
@mwcraig , this seems to do the trick for Python 3.12 but I have no idea if this is correct fix or not. Please review. Thanks!

Reply:
This seems fine -- apparently we expected a `TypeError` to be raised if the uncertainty was unsliceable but now need the `KeyError` too. I didn't the change in Python 3.12 that causes this but the change seems fine.